### ISSUE TITLE
[Platform IOS] My app crashes when show native iOS views in a Flutter application

### ISSUE THREAD

**Issue by MynamesHai** [flutter/flutter#105016]
I use FlutterMethodChannel to transfer data from Flutter to IOS. 
Then register native IOS view show in Flutter.
It worked fine for the first time. But next time the app crashes and doesn't give any logs

* In Dart:
Pass data:
```
 try {
      final String result = await methodChannel.invokeMethod('getDataLevel', {"data": text, "content": content});
    } catch (e) {
    }
```

Show View
```dart
    return UiKitView(
      viewType: viewType,
      layoutDirection: TextDirection.ltr,
      creationParams: creationParams,
      creationParamsCodec: const StandardMessageCodec(),
    );
```

* In Ios - Swift:

![Screen Shot 2022-05-31 at 17 36 02](https://user-images.githubusercontent.com/39751269/171154755-8002cf6b-6800-4b44-84ac-b460570f8d81.png)

```dart
    FirebaseApp.configure()
    GeneratedPluginRegistrant.register(with: self)
    let controller : FlutterViewController = window?.rootViewController as! FlutterViewController
    let videoChannel = FlutterMethodChannel(name: "vast_data",binaryMessenger: controller.binaryMessenger)
    videoChannel.setMethodCallHandler({
           (call: FlutterMethodCall, result: @escaping FlutterResult) -> Void in
                if call.method == "getDataLevel" {
                     guard let args = call.arguments else {
                       return
                     }
                     if let myArgs = args as? [String: Any],
                        let data = myArgs["data"] as? String,
                        let url = myArgs["content"] as? String
                    {
                         url_video = url
                         self.receiveBatteryLevel(result: result, data: data, url: url)
                     } else {
                       result(FlutterError(code: "-1", message: "iOS could not extract " +
                          "flutter arguments in method: (sendParams)", details: nil))
                     }
              }
         })
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
```
And

![Screen Shot 2022-05-31 at 17 35 04](https://user-images.githubusercontent.com/39751269/171154773-b1293187-c262-4200-96c2-22196e1261e4.png)

```swift
private func receiveBatteryLevel(result: FlutterResult, data: String, url: String) {
        // Crash here when second times
        weak var registrar = self.registrar(forPlugin: "plugin-name")
        let factory = FLNativeViewFactory(messenger: registrar!.messenger(),data: "data", url: url)
        self.registrar(forPlugin: "<plugin-name>")!.register(factory, withId: "<platform-view-type>")
        result("Success url_video = \(url)")
    }
```

Thanks

**Comment by maheshmnj** [comment]
Hi @MynamesHai, Thanks for filing the issue. Please share the output of `flutter run -v` (in a textfile as the output will be long) when the crash occurs along with the complete output of `flutter doctor -v`.

**Comment by MynamesHai** [comment]
Hi @maheshmnj , Thanks for the reply. This is out put of `flutter run - v`:
[flutter_run_v.txt](https://github.com/flutter/flutter/files/8810822/flutter_run_v.txt)

and 'flutter doctor -v':
`
[✓] Flutter (Channel unknown, 2.2.2, on macOS 11.3.1 20E241 darwin-arm, locale en-VN)
    • Flutter version 2.2.2 at /Users/buihai/Documents/flutter
    • Framework revision d79295af24 (12 months ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/buihai/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.0, Build version 13A233
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] Connected device (3 available)
    • Redmi Note 7 (mobile)      • 12ecc402                             • android-arm64  • Android 9 (API 28)
    • iPhone 13 Pro Max (mobile) • F8B7E4CA-0F31-4475-B95F-F34CDB0DE663 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 102.0.5005.61

• No issues found!

`

**Comment by maheshmnj** [comment]
Thanks for the update @MynamesHai, You seem to be on a very old version of flutter, There have been lot of changes since that version. I would suggest upgrading to latest stable version and see if the issue is fixed.
You may upgrade by running 

```
flutter channel stable
flutter upgrade -f
``` 
Also I looked at your logs it doesn't seem to show any error and appllication seems to be building fine


>[ +167 ms] Updating files.
[        ] DevFS: Sync finished
[   +7 ms] Syncing files to device iPhone 13 Pro Max... (completed in 258ms)
[   +1 ms] Synced 0.0MB.
[   +1 ms] <- accept
[  +82 ms] Connected to _flutterView/0x7f88fd824820.
[   +6 ms] Flutter run key commands.
[   +2 ms] r Hot reload. 🔥🔥🔥
[   +3 ms] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[        ] An Observatory debugger and profiler on iPhone 13 Pro Max is available at: http://127.0.0.1:58287/X4_rZDlKR24=/
[+1016 ms] The Flutter DevTools debugger and profiler on iPhone 13 Pro Max is available at: http://127.0.0.1:9101?uri=http%3A%2F%2F127.0.0.1%3A58287%2FX4_rZDlKR24%3D%2F
[+9749 ms] flutter: application resumed
[+8942 ms] flutter: widget.urlVideo: https://storage.googleapis.com/gvabox/media/samples/stock.mp4
[+4877 ms] flutter: widget.urlVideo: https://storage.googleapis.com/gvabox/media/samples/stock.mp4
[  +84 ms] Service protocol connection closed.
[        ] Lost connection to device.
[   +3 ms] DevFS: Deleting filesystem on the device
(file:///Users/buihai/Library/Developer/CoreSimulator/Devices/F8B7E4CA-0F31-4475-B95F-F34CDB0DE663/data/Containers/Data/Application/9B815A3E-DE56-4A61-B988-6CBAB2150309/tmp/vnn-system-mobilelvOVxS/vnn-syste
m-mobile/)
[ +266 ms] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed
[  +23 ms] "flutter run" took 106,224ms.
[  +10 ms] ensureAnalyticsSent: 1ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 0
